plugins/chat/lib/chat/statistics.rb

```diff
@@ -41,6 +41,47 @@ def self.about_users
       }
     end
 
+    def self.channel_messages
+      query =
+        Chat::Message.joins(:chat_channel).where.not(chat_channel: { type: "DirectMessageChannel" })
+
+      {
+        :last_day => query.where("chat_messages.created_at > ?", 1.days.ago).count,
+        "7_days" => query.where("chat_messages.created_at > ?", 7.days.ago).count,
+        "30_days" => query.where("chat_messages.created_at > ?", 30.days.ago).count,
+        :count => query.count,
+      }
+    end
+
+    def self.direct_messages
+      query =
+        Chat::Message.joins(:chat_channel).where(chat_channel: { type: "DirectMessageChannel" })
+
+      {
+        :last_day => query.where("chat_messages.created_at > ?", 1.days.ago).count,
+        "7_days" => query.where("chat_messages.created_at > ?", 7.days.ago).count,
+        "30_days" => query.where("chat_messages.created_at > ?", 30.days.ago).count,
+        :count => query.count,
+      }
+    end
+
+    def self.open_channels_with_threads_enabled
+      query = Chat::Channel.where(threading_enabled: true, status: :open)
+
+      { :last_day => 0, "7_days" => 0, "30_days" => 0, :count => query.count }
```

Comment:
Plugin API require us provide all four components here, but for amount of channels with threads enabled the components _last_day_, _7_days_ and _30_days_ doesn't really make sense, so I'm returning zeros here.